src/CTA.WebForms2Blazor/Services/WorkspaceManagerService.cs

```diff
@@ -73,6 +83,17 @@ public ProjectId CreateProjectFile(string projectName, IEnumerable<ProjectRefere
             Project project = _workspace.AddProject(projectInfo);
             _numProjects += 1;
 
+            if (_numProjects >= _expectedProjects)
+            {
+                _allProjectsInWorkspaceTaskSources.ForEach(source =>
+                {
+                    if (!source.Task.IsCanceled)
+                    {
+                        source.SetResult(true);
+                    }
+                });
+            }
```

Comment:
Is there a case where the number of Projects is greater than the expected number of projects?

Reply:
There shouldn't be, I just used >= because it's more robust, but maybe it would be best to throw an exception here